src/Arcus.Security.Providers.AzureKeyVault/KeyVaultCachedSecretProvider.cs

```diff
@@ -0,0 +1,117 @@
+        /// <returns>Returns a <see cref="Secret"/> that contains the latest information for the given secret</returns>
+        /// <exception cref="ArgumentException">The <paramref name="secretName"/> must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The <paramref name="secretName"/> must not be null</exception>
+        /// <exception cref="ArgumentException">The <paramref name="secretValue"/> must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The <paramref name="secretValue"/> must not be null</exception>
+        /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        /// <exception cref="KeyVaultErrorException">The call for a secret resulted in an invalid response</exception>
+        public virtual async Task<Secret> StoreSecretAsync(string secretName, string secretValue)
+        {
+            Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name to request a secret in Azure Key Vault");
+            Guard.NotNullOrWhitespace(secretValue, nameof(secretValue), "Requires a non-blank secret value to store a secret in Azure Key Vault");
+            Guard.For<FormatException>(() => !SecretNameRegex.IsMatch(secretName), "Requires a secret name in the correct format to request a secret in Azure Key Vault, see https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#objects-identifiers-and-versioning");
+
+            Secret secret = await StoreSecretAsync(secretName, secretValue, ignoreCache: false);
+            return secret;
+        }
+
+        /// <summary>
+        /// Stores a secret value with a given secret name
+        /// </summary>
+        /// <param name="secretName">The name of the secret</param>
+        /// <param name="secretValue">The value of the secret</param>
+        /// <param name="ignoreCache">Indicates if the cache should be used or skipped</param>
+        /// <returns>Returns a <see cref="Secret"/> that contains the latest information for the given secret</returns>
+        /// <exception cref="ArgumentException">The <paramref name="secretName"/> must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The <paramref name="secretName"/> must not be null</exception>
+        /// <exception cref="ArgumentException">The <paramref name="secretValue"/> must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The <paramref name="secretValue"/> must not be null</exception>
+        /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        /// <exception cref="KeyVaultErrorException">The call for a secret resulted in an invalid response</exception>
+        public virtual async Task<Secret> StoreSecretAsync(string secretName, string secretValue, bool ignoreCache)
```

Comment:
When storing a secret the cache should always be updated so we can remove this flag here actually; otherwise, we'll be working with stale secrets because somebody ignored the cache

Reply:
Yes, done, updated the signature and related tests. Resolving.